frigate/review/maintainer.py

```diff
@@ -430,15 +441,55 @@ def run(self) -> None:
             # check if there is an updated config
             while True:
                 (
-                    updated_topic,
+                    updated_record_topic,
                     updated_record_config,
-                ) = self.config_subscriber.check_for_update()
+                ) = self.record_config_subscriber.check_for_update()
 
-                if not updated_topic:
+                (
+                    updated_review_topic,
+                    updated_review_config,
+                ) = self.review_config_subscriber.check_for_update()
+
+                if not updated_record_topic and not updated_review_topic:
                     break
 
-                camera_name = updated_topic.rpartition("/")[-1]
-                self.config.cameras[camera_name].record = updated_record_config
+                if updated_record_topic:
+                    camera_name = updated_record_topic.rpartition("/")[-1]
+                    self.config.cameras[camera_name].record = updated_record_config
+
+                if updated_review_topic:
+                    camera_name = updated_review_topic.rpartition("/")[-1]
+                    old_alerts_enabled = self.config.cameras[
+                        camera_name
+                    ].review.alerts.enabled
+                    old_detections_enabled = self.config.cameras[
+                        camera_name
+                    ].review.detections.enabled
+                    self.config.cameras[camera_name].review = updated_review_config
+
+                    # Check if alerts.enabled or detections.enabled has changed
+                    if (
+                        old_alerts_enabled
+                        != self.config.cameras[camera_name].review.alerts.enabled
+                        or old_detections_enabled
+                        != self.config.cameras[camera_name].review.detections.enabled
+                    ):
+                        segment = self.active_review_segments.get(camera_name)
+                        if segment:
+                            if (
+                                not self.config.cameras[
+                                    camera_name
+                                ].review.alerts.enabled
+                                and segment.severity == SeverityEnum.alert
+                            ):
+                                self.end_segment(camera_name)
+                            elif (
+                                not self.config.cameras[
+                                    camera_name
+                                ].review.detections.enabled
+                                and segment.severity == SeverityEnum.detection
+                            ):
+                                self.end_segment(camera_name)
```

Comment:
This seems like duplicate logic to below, I think we can remove this and just rely on the code below